sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -373,13 +373,14 @@ public class ChatClient {
         }
     }
 
-    /// Start push notifications. Receiving of notifications can be expected ~1 second after successfully registering.
+    /// Start push notifications. Receiving of notifications can be expected after successfully registering.
     /// - Parameters:
     ///   - deviceToken: APNS push token.
+    ///   - pushNotificationEncryptionDelegate: Manage encyrption keys used by push notification.
     ///   - completionHandler: Success indicates request to register for notifications has been received.
     public func startPushNotifications(
         deviceToken: String,
-        encryptionKeyPair: EncryptionKeyPair,
+        pushNotificationEncryptionDelegate: PushNotificationEncryptionDelegate?,
```

Comment:
The delegate should not be passed to these calls. It should be set on the client directly.

Reply:
Not sure if I understand this comment correctly. Could you please elaborate more on "Set on the client directly"?

Reply:
Let me clarify my understanding - We provide a protocol "PushNotificationEncryptionDelegate" and the Contoso needs to create a concrete class to adhere this Protocol, for example "AppGroupBasedDelegate". They will need to initialize the "AppGroupBasedDelegate" and pass it into chat SDK. So Chat SDK can call it to "persist key" when starting PN and "get keys" when decrypting PN.

Reply:
Could you please provide a reason on why we should treat it as a "Delegate"? Are there any guidelines? @tjprescott

Reply:
I provided a delegate property on the client side so Contoso can set the delegate as soon as they init the ChatClient. Code change pushed.